packages/forms/frontend/sirius-components-widget-reference/src/components/FilterableSortableList.tsx

```diff
@@ -0,0 +1,195 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+import { ServerContext } from '@eclipse-sirius/sirius-components-core';
+import { splitText } from '@eclipse-sirius/sirius-components-trees';
+import List from '@material-ui/core/List';
+import ListItem from '@material-ui/core/ListItem';
+import ListItemIcon from '@material-ui/core/ListItemIcon';
+import ListItemText from '@material-ui/core/ListItemText';
+import { makeStyles } from '@material-ui/core/styles';
+import Typography from '@material-ui/core/Typography';
+import DragHandleIcon from '@material-ui/icons/DragHandle';
+import React, { useContext, useState } from 'react';
+import { FilterableSortableListProps } from './FilterableSortableList.types';
+import { ModelBrowserFilterBar } from './ModelBrowserFilterBar';
+
+const useStyles = makeStyles((theme) => ({
+  selectable: {
+    cursor: 'pointer',
+    '&:hover': {
+      stroke: '#66808a',
+      fill: '#66808a',
+      backgroundColor: 'var(--blue-lagoon-lighten-95)',
+    },
+  },
+  selected: {
+    cursor: 'pointer',
+    stroke: '#66808a',
+    fill: '#66808a',
+    backgroundColor: 'var(--blue-lagoon-lighten-90)',
+  },
+  selectedLabel: {
+    fontWeight: 'bold',
+  },
+  title: {
+    opacity: 0.6,
+    fontSize: 12,
+  },
+  borderStyle: {
+    border: '1px solid #0000008A',
+    height: 300,
+    overflow: 'auto',
+  },
+  highlight: {
+    backgroundColor: theme.palette.navigation.leftBackground,
+  },
+}));
+
+const highlightText = (label: string, textToHighlight: string, classes): JSX.Element => {
```

Comment:
`classes` is not typed

Reply:
A function which returns from JSX is a component, it should be treated as such (name, props, etc)

Reply:
I turned it into a component, not 100% sure of the benefit of releasing it as a separate component.